### PR TITLE
Support installing vte, gtk+, pygtk, py2cairo, libressl

### PR DESCRIPTION
**Pull request by paulhybryant**
Support installing vte, gtk+, pygtk, py2cairo, libressl

### CODE REVIEW

File: Library/Formula/gtk+.rb [hunk 1]
```diff
@@ -12,7 +12,9 @@ class Gtkx < Formula
     sha256 "3f3948ce65bd2c199e2603a8bebc39f6de14e3c7f408226b1d6a0c8b52a51f29" => :mountain_lion
   end
 
-  option "with-quartz-relocation", "Build with quartz relocation support"
+  if OS.mac?
+    option "with-quartz-relocation", "Build with quartz relocation support"
+  end
```
Comment (sjackman) [inline Library/Formula/gtk+.rb]: Please remove this chunk. I prefer a minimall patch, to make merging with Homebrew easier.

File: Library/Formula/gtk+.rb [hunk 2]
```diff
@@ -33,10 +35,14 @@ def install
             "--prefix=#{prefix}",
             "--disable-glibtest",
             "--enable-introspection=yes",
-            "--with-gdktarget=quartz",
```
Comment (sjackman) [inline Library/Formula/gtk+.rb]: Use instead
```ruby
"--with-gdktarget=#{OS.mac? ? 'quartz' : 'x11'}",
```

File: Library/Formula/gtk+.rb [hunk 3]
```diff
@@ -33,10 +35,14 @@ def install
...
             "--disable-visibility"]
 
-    args << "--enable-quartz-relocation" if build.with?("quartz-relocation")
+    if OS.mac?
+      args << "--enable-gdktarget=quartz"
+      args << "--enable-quartz-relocation" if build.with?("quartz-relocation")
+    else
+      args << "--enable-gdktarget=x11"
+    end
```
Comment (sjackman) [inline Library/Formula/gtk+.rb]: Please remove this chunk.

File: Library/Formula/gtk+.rb [hunk 4]
```diff
@@ -86,16 +92,27 @@ def install
       -L#{pango.opt_lib}
       -latk-1.0
       -lcairo
-      -lgdk-quartz-2.0
```
Comment (sjackman) [inline Library/Formula/gtk+.rb]: Please use inline `#{` as above.

**Comment by sjackman** [comment]
Please patch each formula in a separate pull request.
See https://github.com/Homebrew/linuxbrew/blob/master/CONTRIBUTING.md

File: Library/Formula/py2cairo.rb [hunk 5]
```diff
@@ -40,7 +40,11 @@ def install
     system "./waf", "install"
 
     module_dir = lib/"python2.7/site-packages/cairo"
-    mv module_dir/"lib_cairo.dylib", module_dir/"_cairo.so"
+    if OS.mac?
+      mv module_dir/"lib_cairo.dylib", module_dir/"_cairo.so"
+    else
+      mv module_dir/"lib_cairo.so", module_dir/"_cairo.so"
+    end
```
Comment (sjackman) [inline Library/Formula/py2cairo.rb]: Please use
```ruby
dylib = OS.mac? ? "dylib" : "so"
mv module_dir/"lib_cairo.#{dylib}", module_dir/"_cairo.so"
```

File: Library/Formula/vte.rb [hunk 6]
```diff
@@ -77,17 +77,28 @@ def install
       -L#{pango.opt_lib}
       -latk-1.0
       -lcairo
-      -lgdk-quartz-2.0
```
Comment (sjackman) [inline Library/Formula/vte.rb]: Please use inline `#{…}`.

**Comment by sjackman** [comment]
Thanks for your contribution, Paul. Please submit the fix for each formula in its own pull request. The commit log for each pull request should be:
```
foo 1.2.3: Fix for Linuxbrew

Fix the error: quote the exact error message here
```

**Comment by paulhybryant** [comment]
Will create separate pull request.

**Comment by sjackman** [comment]
Thanks, Paul.

**Pull request closed without merging**